src/hooks/hook_area02.nss

```diff
@@ -15,8 +15,20 @@ void main()
     // Don't run this event if the exiting object is a PC that is about to be
     // booted.
     object oPC = GetExitingObject();
-    if (GetIsPC(oPC) && GetLocalInt(oPC, LOGIN_BOOT))
-        return;
+
+    if (HasListObject(OBJECT_SELF, oPC, AREA_ROSTER))
+    {
+        if (GetLocalInt(oPC, LOGIN_BOOT))
+            return;
+
+        if (!RemoveListObject(OBJECT_SELF, oPC, AREA_ROSTER) 
+                && ENABLE_ON_AREA_EMPTY_EVENT
+                && !CountObjectList(OBJECT_SELF, AREA_ROSTER))
```

Comment:
This line is unnecessary since `RemoveListObject()` returns the updated list count.

Reply:
Good call, keep forgetting about that.  Every time I think I'm familiar with these procedures, I keep finding out I'm not.  It's been removed.  Also, it appears the `HasListObject` is also superfluous, so that was also removed.  Doesn't matter if the oPC object was on the list or not, it's only there to check to see if the timer needs to be set, and the timer should be set even if there was some kind of error and the oPC wasn't on the AREA_ROSTER.